languages/java/exercises/concept/conditionals-ternary/.docs/after.md

```diff
@@ -0,0 +1,33 @@
+_ternary operators_ can be thought of as being _if-else_ inline. It is a good example that most of the times you have different ways of achieving the same result.
+
+So, how to decide between _if-else_ and _ternary_ ? Well, _Ternary_ is used in simple cenarios, where you just need to return a value based on a condition and no extra computation is needed, use _if-else_ for everthing else, like nested conditions, big expressions and when more than one line is needed to decide the return value.
+
+And try to never nest _ternary operators_ as the expression becomes illegible, prefer nested if's to handle those cases
+
+```java
+
+// hard to read
+int value = expr1 ? expr2 ? val1 : expr3 ? val2 : val3 : val4
+
+// easier to read
+if (expr1){
+    if (expr2){
+        return val1;
+    } else{
+        if (expr3) {
+            return val2;
+        } else {
+            return val3;
+        }
+    }
+} else {
+    return val4;
+}
```

Comment:
I'm no Java expert, but I find this not easier to read :)

```suggestion
// easier to read
if (expr1){
    if (expr2){
        return val1;
    } else{
        if (expr3) {
            return val2;
        } else {
            return val3;
        }
    }
} else {
    return val4;
}
```

Reply:
Unfortunately i understand you, i think most of this is due to the brackets.
I think a shallow nested if will do the trick

Reply:
👍

Reply:
```java
// hard to read
int value = expr1 ? expr2 && expr3 ? val1 : (val2 + val3) : val4;

// easier to read
if (expr1){

    if (expr2 && expr3){

        return val1;
    } else {

        return val2 + val3;
    }

} else {
    
    return val4;
}

 ```